docs/Getting Started/Debian/Debian Bullseye Root on ZFS.rst

```diff
@@ -0,0 +1,1117 @@
+
+        grub-install $DISK
+
+   Note that you are installing GRUB to the whole disk, not a partition.
+
+   If you are creating a mirror or raidz topology, repeat the ``grub-install``
+   command for each disk in the pool.
+
+   #. For UEFI booting, install GRUB to the ESP::
+
+        grub-install --target=x86_64-efi --efi-directory=/boot/efi \
+            --bootloader-id=debian --recheck --no-floppy
+
+      It is not necessary to specify the disk here. If you are creating a
+      mirror or raidz topology, the additional disks will be handled later.
+
+#. Fix filesystem mount ordering:
+
+   We need to activate ``zfs-mount-generator``. This makes systemd aware of
+   the separate mountpoints, which is important for things like ``/var/log``
+   and ``/var/tmp``. In turn, ``rsyslog.service`` depends on ``var-log.mount``
+   by way of ``local-fs.target`` and services using the ``PrivateTmp`` feature
+   of systemd automatically use ``After=var-tmp.mount``.
+
+   ::
+
+     mkdir /etc/zfs/zfs-list.cache
+     touch /etc/zfs/zfs-list.cache/bpool
+     touch /etc/zfs/zfs-list.cache/rpool
+     ln -s /usr/lib/zfs-linux/zed.d/history_event-zfs-list-cacher.sh /etc/zfs/zed.d
+     zed -F &
+
+   Verify that ``zed`` updated the cache by making sure these are not empty::
+
+     cat /etc/zfs/zfs-list.cache/bpool
+     cat /etc/zfs/zfs-list.cache/rpool
+
+   If either is empty, force a cache update and check again::
+
+     zfs set canmount=on     bpool/BOOT/debian
+     zfs set canmount=noauto rpool/ROOT/debian
+
+   If they are still empty, stop zed (as below), start zed (as above) and try
+   again.
+
+   Stop ``zed``::
+
+     fg
+     Press Ctrl-C.
+
+   Fix the paths to eliminate ``/mnt``::
+
+     sed -Ei "s|/mnt/?|/|" /etc/zfs/zfs-list.cache/*
+
+Step 6: First Boot
+------------------
+
+#. Optional: Install SSH::
+
+     apt install --yes openssh-server
+
+     vi /etc/ssh/sshd_config
+     # Set: PermitRootLogin yes
+
+#. Optional: Snapshot the initial installation::
+
+     zfs snapshot bpool/BOOT/debian@install
+     zfs snapshot rpool/ROOT/debian@install
+
+   In the future, you will likely want to take snapshots before each
+   upgrade, and remove old snapshots (including this one) at some point to
+   save space.
+
+#. Exit from the ``chroot`` environment back to the LiveCD environment::
+
+     exit
+
+#. Run these commands in the LiveCD environment to unmount all
+   filesystems::
+
+     mount | grep -v zfs | tac | awk '/\/mnt/ {print $3}' | \
+         xargs -i{} umount -lf {}
+     zpool export -a
+
+#. Reboot::
+
+     reboot
+
+   Wait for the newly installed system to boot normally. Login as root.
+
+#. Create a user account:
+
+   Replace ``username`` with your desired username::
+
+     zfs create rpool/home/username
+     adduser username
+
+     cp -a /etc/skel/. /home/username
+     chown -R username:username /home/username
+     usermod -a -G audio,cdrom,dip,floppy,netdev,plugdev,sudo,video username
```

Comment:
I propose to leave a variant from Buster with username in variable to minimize client's changes

```suggestion
   Replace ``YOUR_USERNAME`` with your desired username::

     username=YOUR_USERNAME

     zfs create rpool/home/$username
     adduser $username

     cp -a /etc/skel/. /home/$username
     chown -R $username:$username /home/$username
     usermod -a -G audio,cdrom,dip,floppy,netdev,plugdev,sudo,video $username
```